airbyte-cdk/python/airbyte_cdk/sources/concurrent_source/concurrent_read_processor.py

```diff
@@ -62,7 +61,7 @@ def __init__(
         self._streams_done: Set[str] = set()
         self._exceptions_per_stream_name: dict[str, List[Exception]] = {}
 
-    def on_partition_generation_completed(self, sentinel: PartitionGenerationCompletedSentinel) -> Iterable[AirbyteMessage]:
+    def on_partition_generation_completed(self, sentinel: PartitionGenerationCompletedSentinel) -> Iterable[AirbyteMessage | None]:
```

Comment:
does the iterable really include `None`s? This sounds difficult to work with

Reply:
`mypy` raises an error due to `start_next_partition_generator`, that can return `None`
I did not really dig into this class, only fixed `mypy`. I suppose we'll refactor this much during the next project: concurrency for Low-Code

https://github.com/airbytehq/airbyte/blob/feded1fd2f9006326e3d62ffede6c919492f8998/airbyte-cdk/python/airbyte_cdk/sources/concurrent_source/concurrent_read_processor.py#L160-L179

Reply:
if it's just for mypy, let's add a type ignore. Otherwise, the type will indicate that it is valid to yield None, which might not be the case?

Reply:
ignored